core/server/common/src/main/java/alluxio/master/journal/JournalEntryAssociation.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.journal;
+
+import alluxio.Constants;
+import alluxio.exception.InvalidJournalEntryException;
+import alluxio.proto.journal.Journal.JournalEntry;
+
+/**
+ * Association from journal entry to the master it applies to.
+ */
+public final class JournalEntryAssociation {
+
+  /**
+   * @param entry a journal entry
+   * @return the name of the master responsible for the given journal entry
+   * @throws InvalidJournalEntryException if the journal entry is unrecognized
+   */
+  public static String getMasterForEntry(JournalEntry entry) throws  InvalidJournalEntryException {
```

Comment:
It's difficult because our journaling infrastructure is in `core-server-common`, but the specific masters are in `core-server-master`, which depends on `core-server-common`. `core-server-common` has access to the protobuf definitions of all entries, and to the Constants used to define master names, but not to the specific master implementations.

We can address the issue of forgetting to update the file by having a unit test which iterates over all types of journal entries and makes sure each is associated with a master name. Do you have any further concerns with having the journal entry association in `core-server-common`?